presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveTransactionalTable.java

```diff
@@ -166,4 +182,66 @@ private static String hiveTableProperties(TransactionalTableType transactionalTa
                 bucketingType.getHiveTableProperties().stream())
                 .collect(joining(",", "TBLPROPERTIES (", ")"));
     }
+
+    private static void compactTableAndWait(CompactionMode compactMode, String tableName, Duration timeout)
+    {
+        onHive().executeQuery(format("ALTER TABLE %s COMPACT '%s'", tableName, compactMode.name()));
+
+        Set<String> tableCompactionsIds = getTableCompactions(compactMode, tableName)
+                .map(row -> row.get("compactionid"))
+                .collect(toUnmodifiableSet());
+
+        assertEventually(timeout, () -> {
+            Set<String> completedCompactionsIds = getTableCompactions(compactMode, tableName)
+                    .filter(row -> row.get("state").equals("succeeded"))
+                    .map(row -> row.get("compactionid"))
+                    .collect(toUnmodifiableSet());
+
+            assertTrue(completedCompactionsIds.containsAll(tableCompactionsIds));
+        });
+    }
+
+    private static Stream<Map<String, String>> getTableCompactions(CompactionMode compactionMode, String tableName)
+    {
+        return Stream.of(onHive().executeQuery("SHOW COMPACTIONS")).flatMap(TestHiveTransactionalTable::mapRows)
+                .filter(row -> isTableCompaction(compactionMode, tableName, row));
+    }
+
+    private static Stream<Map<String, String>> mapRows(QueryResult result)
+    {
+        if (result.getRowsCount() == 0) {
+            return Stream.of();
+        }
+
+        List<?> columnNames = result.row(0).stream()
+                .filter(Objects::nonNull)
+                .collect(toUnmodifiableList());
+
+        ImmutableList.Builder<Map<String, String>> rows = ImmutableList.builder();
+        for (int rowIndex = 1; rowIndex < result.getRowsCount(); rowIndex++) {
```

Comment:
why 1?

Reply:
because first row contains column names (don't ask me why):

```
0: jdbc:hive2://localhost:10000/default> show compactions;
INFO  : Compiling command(queryId=hive_20200616160810_8658fc21-d5df-4028-9955-84af206d7d6a): show compactions
INFO  : Semantic Analysis Completed (retrial = false)
INFO  : Created Hive schema: Schema(fieldSchemas:[FieldSchema(name:compactionid, type:string, comment:from deserializer), FieldSchema(name:dbname, type:string, comment:from deserializer), FieldSchema(name:tabname, type:string, comment:from deserializer), FieldSchema(name:partname, type:string, comment:from deserializer), FieldSchema(name:type, type:string, comment:from deserializer), FieldSchema(name:state, type:string, comment:from deserializer), FieldSchema(name:hostname, type:string, comment:from deserializer), FieldSchema(name:workerid, type:string, comment:from deserializer), FieldSchema(name:starttime, type:string, comment:from deserializer), FieldSchema(name:duration, type:string, comment:from deserializer), FieldSchema(name:hadoopjobid, type:string, comment:from deserializer), FieldSchema(name:errormessage, type:string, comment:from deserializer)], properties:null)
INFO  : Completed compiling command(queryId=hive_20200616160810_8658fc21-d5df-4028-9955-84af206d7d6a); Time taken: 1.118 seconds
INFO  : Executing command(queryId=hive_20200616160810_8658fc21-d5df-4028-9955-84af206d7d6a): show compactions
INFO  : Starting task [Stage-0:DDL] in serial mode
INFO  : Completed executing command(queryId=hive_20200616160810_8658fc21-d5df-4028-9955-84af206d7d6a); Time taken: 0.059 seconds
INFO  : OK
+---------------+-----------+----------+------------+-------+--------+-----------+-------------+---------------+--------------+----------------+---------------+
| compactionid  |  dbname   | tabname  |  partname  | type  | state  | hostname  |  workerid   |   starttime   |   duration   |  hadoopjobid   | errormessage  |
+---------------+-----------+----------+------------+-------+--------+-----------+-------------+---------------+--------------+----------------+---------------+
| CompactionId  | Database  | Table    | Partition  | Type  | State  | Worker    | Start Time  | Duration(ms)  | HadoopJobId  | Error message  | NULL          |
+---------------+-----------+----------+------------+-------+--------+-----------+-------------+---------------+--------------+----------------+---------------+
1 row selected (1.512 seconds)
```